pkg/api/types.go

```diff
@@ -1801,6 +1801,15 @@ type SecretType string
 
 const (
 	SecretTypeOpaque SecretType = "Opaque" // Default; arbitrary user-defined data
+
+	// SecretTypeDockerConfig contains a dockercfg file that follows the same format rules as ~/.dockercfg
+	//
+	// Required fields:
+	// - Secret.Data["dockercfg"] - a serialized ~/.dockercfg file
+	SecretTypeDockerConfig SecretType = "DockerConfig"
```

Comment:
just occurred to me that we may want to qualify secret types (e.g. prefix with "kubernetes.io/")